File: Assets/js/TextShadow.js
#pragma strict

import UnityEngine.UI;

var originalText : Text;
var thisText : Text;

function Awake () {
	var rt : RectTransform = GetComponent(RectTransform);
	originalText = rt.parent.GetComponent(Text);
	thisText = GetComponent(Text);
}

function Start () {
}

function Update () {
	if (originalText.enabled != thisText.enabled) {
		thisText.enabled = originalText.enabled;
	}
	
	if (originalText.text != thisText.text) {
		thisText.text = originalText.text;
	}
	
	if (originalText.color.a != thisText.color.a) {
		thisText.color.a = originalText.color.a;
	}
}

File: Assets/cs/TextShadow.cs
using UnityEngine;
using System;
using UnityEngine.UI;


public class TextShadow:MonoBehaviour{
	
	public Text originalText;
	public Text thisText;
	
	public void Awake() {
		RectTransform rt = GetComponent<RectTransform>();
		originalText = rt.parent.GetComponent<Text>() as Text;
		thisText = GetComponent<Text>() as Text;
	}
	
	public void Start() {
	}
	
	public void Update() {
		if (originalText.enabled != thisText.enabled) {
			thisText.enabled = originalText.enabled;
		}
		
		if (originalText.text != thisText.text) {
			thisText.text = originalText.text;
		}
		
		if (originalText.color.a != thisText.color.a) {
			var tmp_cs1 = thisText.color;
            tmp_cs1.a = originalText.color.a;
            thisText.color = tmp_cs1;
		}
	}

}